.github/workflows/sync_default_branch.yml

```diff
@@ -0,0 +1,27 @@
+name: Sync Default Branch with Target Branch
+
+on:
+  workflow_call:
+    inputs:
+      target_branch:
+        required: false
+        type: string
+        default: storybook-site
+    secrets:
+      caller_github_token:
+        required: true
+
+jobs:
+  sync-branches:
+    runs-on: ubuntu-latest
+    name: Sync branches
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Sync ${{ github.event.repository.default_branch }} with ${{ inputs.target_branch }}
+        uses: devmasx/merge-branch@v1.4.0
+        with:
+          type: now
+          from_branch: ${{ github.event.repository.default_branch }}
+          target_branch: ${{ inputs.target_branch }}
+          github_token: ${{ secrets.caller_github_token }}
```

Comment:
"react-components" and "search-ui-react" use `secrets.GITHUB_TOKEN` currently, and the other secrets like percy are all caps. Can we continue this

Reply:
this is consistent with the caller github token in our other workflows like `coverage`, `version-update`, and `sync_develop_and_main`. I think it would be better to be consistent, and changing the naming for those workflows would be a breaking change